pkg/controllers/customeradmin/controllernamespace/namespace/namespace_controller.go

```diff
@@ -0,0 +1,126 @@
+package namespace
+
+import (
+	"context"
+	"strings"
+
+	rbs "github.com/openshift/openshift-azure/pkg/controllers/customeradmin/rolebindings"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// Add creates a new CustomerAdminReconciler Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &reconcileNamespace{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("customeradmin-namespace-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(&source.Kind{Type: &corev1.Namespace{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &reconcileNamespace{}
+
+// ReconcileNamespace reconciles a CustomerAdminReconciler object
+type reconcileNamespace struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile receives a namespace event and reconciles rolebindings for
+// customer roles
+// Note:
+// The Controller will requeue the Request to be processed again if the
+// returned error is non-nil or Result.Requeue is true, otherwise upon
+// completion it will remove the work from the queue.
+func (r *reconcileNamespace) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	for _, forbiddenName := range []string{"default", "kube", "openshift", "kubernetes"} {
```

Comment:
Still wrong! Copy the logic at https://github.com/openshift/origin/blob/master/pkg/project/apiserver/registry/projectrequest/delegated/delegated.go#L100-L126 and add `default` for good measure.

Reply:
Added the logic from origin + default.